test/Worker/AbstractPoolErrorTest.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Amp\Parallel\Test\Worker;
+
+use Amp\Loop;
+use Amp\Parallel\Worker\Pool;
+use Amp\PHPUnit\TestCase;
+
+abstract class AbstractPoolErrorTest extends TestCase {
+    /**
+     * @param int $min
+     * @param int $max
+     *
+     * @return \Amp\Parallel\Worker\Pool
+     */
+    abstract protected function createPool($max = Pool::DEFAULT_MAX_SIZE): Pool;
+
+    /**
+     * @expectedException        \Error
+     * @expectedExceptionMessage Maximum size must be a non-negative integer
+     */
+    public function testCreatePoolShouldThrowError() {
```

Comment:
Please use `$this->expectException()` instead of annotations.

Reply:
I just follow some places that use the annotations.
Please see this [link](https://github.com/amphp/parallel/blob/master/test/Worker/JobTest.php#L16).

Reply:
I know that there are other places that use annotations, but IMO all new code should use code instead. It doesn't happen often, but it helps with exception renames and also with inherited tests that are overwritten to add something in a child class.